lib/wibox/layout/overflow.lua

```diff
@@ -0,0 +1,549 @@
+
+    -- Set the direction covered by scrolling to the maximum value
+    -- to allow widgets to take as much space as they want.
+    if is_y then
+        height = math.huge
+    else
+        width = math.huge
+    end
+
+    -- First, determine widget sizes.
+    -- Only when the content doesn't fit and needs scrolling should
+    -- we reduce content size to make space for a scrollbar.
+    for _, widget in ipairs(widgets) do
+        local w, h = base.fit_widget(self, context, widget, width, height)
+
+        if is_y then
+            used_max = math.max(used_max, w)
+            used_in_dir = used_in_dir + h
+        else
+            used_max = math.max(used_max, h)
+            used_in_dir = used_in_dir + w
+        end
+    end
+
+    local spacing = self._private.spacing * (num_widgets - 1)
+    used_in_dir = used_in_dir + spacing
+
+    local need_scrollbar = scrollbar_enabled and used_in_dir > avail_in_dir
+
+    -- Even if `used_max == orig_(width|height)` already, `base.fit_widget`
+    -- will clamp return values, so we can "overextend" here.
+    if need_scrollbar then
+        used_max = used_max + scrollbar_width
+    end
+
+    if is_y then
+        return used_max, used_in_dir
+    else
+        return used_in_dir, used_max
+    end
+end
+
+-- Layout children, scrollbar and spacing widgets.
+-- Only those widgets that are currently visible will be placed.
+function overflow:layout(context, orig_width, orig_height)
+    local result = {}
+    local is_y = self._private.dir == "y"
+    local widgets = self._private.widgets
+    local avail_in_dir = is_y and orig_height or orig_width
+    local scrollbar_width = self._private.scrollbar_width
+    local scrollbar_enabled = self._private.scrollbar_enabled
+    local scrollbar_position = self._private.scrollbar_position
+    local width, height = orig_width, orig_height
+    local widget_x, widget_y = 0, 0
+    local used_in_dir, used_max = 0, 0
+
+    -- Set the direction covered by scrolling to the maximum value
+    -- to allow widgets to take as much space as they want.
+    if is_y then
+        height = math.huge
+    else
+        width = math.huge
+    end
+
+    -- First, determine widget sizes.
+    -- Only when the content doesn't fit and needs scrolling should
+    -- we reduce content size to make space for a scrollbar.
+    for _, widget in pairs(widgets) do
+        local w, h = base.fit_widget(self, context, widget, width, height)
+
+        if is_y then
+            used_max = math.max(used_max, w)
+            used_in_dir = used_in_dir + h
+        else
+            used_max = math.max(used_max, h)
+            used_in_dir = used_in_dir + w
+        end
+    end
+
+    used_in_dir = used_in_dir + self._private.spacing * (#widgets-1)
+
+    -- Save size for scrolling behavior
+    self._private.avail_in_dir = avail_in_dir
+    self._private.used_in_dir = used_in_dir
+
+    local need_scrollbar = used_in_dir > avail_in_dir and scrollbar_enabled
```

Comment:
Does that mean that if the content is smaller than the available space, there is no scrollbar even if we explicitly set `scrollbar_enabled = true`? Is there a way to force scrollbar?

Reply:
> there is no scrollbar even if we explicitly set `scrollbar_enabled = true`?
Yes

> Is there a way to force scrollbar?
No